install_files/ansible-base/roles/tails-config/files/securedrop@securedrop.freedom.press/extension.js

```diff
@@ -0,0 +1,129 @@
+/* exported init */
+
+const GETTEXT_DOMAIN = 'securedrop';
+
+const {Clutter, GObject, GLib, St } = imports.gi;
+
+const Util = imports.misc.util;
+const ExtensionUtils = imports.misc.extensionUtils;
+const Main = imports.ui.main;
+const PanelMenu = imports.ui.panelMenu;
+const PopupMenu = imports.ui.popupMenu;
+
+const _ = ExtensionUtils.gettext;
+
+const Indicator = GObject.registerClass(
+class Indicator extends PanelMenu.Button {
+    _init() {
+        super._init(0.0, 'SecureDrop');
+
+        // Check for SSH keys - if they don't exist, this is a Journalist workstation
+        // (and therefore the admin options should remain hidden)
+        let is_admin = false;
+        let [ok, out, err, exit] = GLib.spawn_command_line_sync('/bin/bash -c "test -f ~/Persistent/securedrop/install_files/ansible-base/app-ssh.auth_private"');
```

Comment:
This doesn't cover the (admittedly super-rare) case where the workstation is set up for SSH-over-local, as the `.auth_private` file will not exist.

(I don't think this is a requirement for release, as a: SSH-over-local is bad and wrong in the vast majority of cases and b: said users can still SSH in as before.)

Reply:
I changed this to look for `tor_v3_keys.json` rather than `app-ssh.auth_private`, which should cover the case where a workstation is set up for SSH-over-local.